### PR TITLE
 Bug fixes

### PR DESCRIPTION
**Pull request by nwithan8**
- Fix bug where HTTP error data is not being stored as expected in cassette
  - This is a breaking change (although the feature it breaks likely never worked in the first place)
- Fix bug where expiration time frames other than "forever" and "never" would throw a NullPointerException.
- Bump version to 0.4.2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
 
 ## v0.4.0 (2022-10-04)
 
-- New feature: Set expiration time for interactions (how long since it was recorded should an interaction be considered valid)
-  - Can determine what to do if a matching interaction is considered invalid:
-    - Warn the user, but proceed with the interaction
-    - Throw an exception
-    - Automatically re-record (cannot be used in `Replay` mode)
-- New feature: Pass in a custom Logger instance to EasyVCR to funnel log messages into your own logging setup (fallback: logs to console)
+- New feature: Set expiration time for interactions (how long since it was recorded should an interaction be considered
```
Comment (Justintime50) [inline CHANGELOG.md:18]: Let's leave the previous formatting alone for the CHANGELOG

**Review by Justintime50 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nwithan8) [reply CHANGELOG.md:18]: Bad merge, will fix.

**Review by Justintime50 — APPROVED** [review]

File: src/main/java/com/easypost/easyvcr/TimeFrame.java [hunk 2]
```diff
@@ -89,12 +89,18 @@ public boolean hasLapsed(long fromTimeEpochTimestamp) {
...
      */
     private Instant timePlusFrame(Instant fromTime) {
+        // We need to do a null check here. The "default" case in the switch statement below doesn't handle null enums.
+        if (commonTimeFrame == null) {  // No common time frame was used
+            return fromTime.plus(days, ChronoUnit.DAYS).plus(hours, ChronoUnit.HOURS)
+                    .plus(minutes, ChronoUnit.MINUTES).plus(seconds, ChronoUnit.SECONDS);
+        }
         switch (commonTimeFrame) {
             case Forever:
                 return Instant.MAX;  // will always been in the future
```
Comment (Justintime50) [inline src/main/java/com/easypost/easyvcr/TimeFrame.java:99]: > will always been

Wut 😂

File: src/main/java/com/easypost/easyvcr/TimeFrame.java [hunk 3]
```diff
@@ -89,12 +89,18 @@ public boolean hasLapsed(long fromTimeEpochTimestamp) {
...
             case Never:
                 return Instant.MIN; // will always been in the past
```
Comment (Justintime50) [inline src/main/java/com/easypost/easyvcr/TimeFrame.java:101]: > will always been

Wut 😂

**Review by Justintime50 — APPROVED** [review]

File: src/main/java/com/easypost/easyvcr/TimeFrame.java (hunk 2 above)
Comment (nwithan8) [reply src/main/java/com/easypost/easyvcr/TimeFrame.java:99]: 😐

**Pull request merged**